### PR TITLE
types(ModalSubmitFields): components is an array

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**

This PR simply fixes an issue in ModalSubmitFields#components as it wasn't marked as an array and instead was simply an action row

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by ImRodry** [comment]
Oh cool you're using teams now, just fyi github only officially assigns 2 members of the team at a time despite you all appearing on the side

**Review by kyranet — APPROVED** [review]

**Review by kyranet — DISMISSED** [review]
Upon further inspection, I just noticed there's a mismatch in the types used. In the JS code, we use a custom interface (documented [here](https://old.discordjs.dev/#/docs/discord.js/main/typedef/ActionRowModalData)):

https://github.com/discordjs/discord.js/blob/e04801bba14ce2eb3a285f82449ae2ad051fbd8f/packages/discord.js/src/structures/ModalSubmitFields.js#L12-L16

`ActionRowModalData` is defined in the typings here:

https://github.com/discordjs/discord.js/blob/e04801bba14ce2eb3a285f82449ae2ad051fbd8f/packages/discord.js/typings/index.d.ts#L2218-L2221

However, this is using `ActionRow<ModalActionRowComponent>` instead.

**Comment by ImRodry** [comment]
I didn't notice but yeah the type is wrong however I don't think that one is right either because that one technically indicates that you can chain `components.components` infinitely, despite the actual TextInputComponent object coming with a `components` property set to undefined. Should I just use that type and ignore this?

**Comment by kyranet** [comment]
The types should match what the JS code is, unless the JS code is somehow wrong and has been for a year.

**Comment by ImRodry** [comment]
The only thing "wrong" about the JS code is the fact that components is set as undefined on text input objects. I'll try to modify the types a little to match that

**Comment by ImRodry** [comment]
Also I know how to fix that issue, can I include that in this PR?

**Comment by ImRodry** [comment]
Pushed both the types change and a small change to the transformComponent function that should get rid of all the undefined properties to align it with the data returned by the API and the TS types. If you want me to remove that commit or move it to a different PR lmk

**Review by vladfrangu — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/ModalSubmitInteraction.js [hunk 1]
```diff
@@ -88,12 +88,13 @@ class ModalSubmitInteraction extends BaseInteraction {
    * @returns {ModalData[]}
```
[the remark this answers is not in the document]
Comment (ImRodry) [reply packages/discord.js/src/structures/ModalSubmitInteraction.js:88]: I did not touch that. If it wasn’t private before I won’t make it private. Idk if I should fix this either

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/structures/ModalSubmitInteraction.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jiralite) [reply packages/discord.js/src/structures/ModalSubmitInteraction.js:88]: This can be addressed in another pull request.

**Pull request merged**